File: csharp/DesignPattern.Interpreter/AddExpression.cs

using System.Collections.Generic;
namespace DesignPattern.Interpreter
{
    public class AddExpression : OperatorExpression
    {
        public AddExpression(Expression left, Expression right)
            : base(left, right)
        { 
        
        }

        public override double Interpret(Context context)
        {
            return this.left.Interpret(context) + this.right.Interpret(context);
        }
    }
}

File: c++/InterpreterPattern/AddExpression.cpp
#include "stdafx.h"
#include "AddExpression.h"


AddExpression::AddExpression(Expression *left,Expression *right) : OperatorExpression(left,right)
{

}


double AddExpression::Interpret(Context context)
{
	return this->left->Interpret(context) + this->right->Interpret(context);
}
